### ISSUE TITLE
Chaining rules should raise an error

### ISSUE THREAD

**Issue by davidpelaez** [dry-rb/dry-validation#240]
Hello.

I have a strange situation where the order in which I chain methods in a schema attribute changes the result:

``` ruby
SPECIAL_TOKENS = ["test_approval", "test_denial"]

# Consider the following different possible definitions of the same attribute 
# inside a schema. The schema is called with: {private_token: "test_approval"}

# these two work without error with the indicated valid input
required(:private_token).value(inclusion?: SPECIAL_TOKENS)
required(:private_token).value(inclusion?: SPECIAL_TOKENS).str?

# But this the following gives: (...) 
# messages={ :private_token=>[ "must be a string", 
# "must be one of: test_approval, test_denial" ]
required(:private_token).str?.value(inclusion?: SPECIAL_TOKENS)
```

The only difference is that in some cases `str?` comes in the end or is absent because it would be implied in the inclusion validation. However I don't see why they would fail in the third case, `str?` should pass and the inclusion validation should pass as well. 

Am I missing a usage rule here or is this indeed a bug where order shouldn't be meaningful in this explicit case and something's unexpectedly colliding somewhere?

**Comment by solnic** [comment]
Sorry but chaining is actually not supported, we just need to intercept it and raise an error. I'll leave this open for a reference and close it once I improve the DSL so that it doesn't allow chaining

**Comment by solnic** [comment]
I should mention that you can simply do `value(inclusion: SPECIAL_TOKENS)` as `str?` is pretty much redundant in such cases

**Comment by davidpelaez** [comment]
Thanks for the clarification, I did indeed feel that `str?` as redudant. I updated the title of the issue to reflect the actual issue.

**Comment by flash-gordon** [comment]
We indeed improved the DSL, this will raise an error in the next version.
